ddsenabler/test/ddsEnablerTests/ReloadConfig.cpp

```diff
@@ -0,0 +1,198 @@
+public:
+    using DDSEnabler::DDSEnabler;
+    const void get_allowed_topics(std::shared_ptr<ddspipe::core::AllowedTopicList>& ptr) const {
+        ptr = std::make_shared<ddspipe::core::AllowedTopicList>(
+            this->configuration_.ddspipe_configuration.allowlist,
+            this->configuration_.ddspipe_configuration.blocklist); 
+    }
+};
+
+// Test the configuration reload when the json configuration file is modified
+TEST(ReloadConfig, json)
+{
+    auto configfile = "./file_watcher_test.json";
+    write_json_file(configfile, 0);
+    
+    // Create DDS Enabler
+    std::unique_ptr<DDSEnabler> enabler;
+    bool result = create_dds_enabler(configfile, data_callback, type_callback, nullptr, enabler);
+    ASSERT_TRUE(result);
+
+    // Create DDSEnablerAccessor to access protected configuration
+    auto enabler_accessor = static_cast<DDSEnablerAccessor*>(enabler.get());
+
+    // Take initial configuration (allowed topics)
+    std::shared_ptr<ddspipe::core::AllowedTopicList> allowed_topics_init;
+    enabler_accessor->get_allowed_topics(allowed_topics_init);
+    
+    // Modify configuration file
+    write_json_file(configfile, 1);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    // Take final configuration (allowed topics)
+    std::shared_ptr<ddspipe::core::AllowedTopicList> allowed_topics_final;
+    enabler_accessor->get_allowed_topics(allowed_topics_final);
+
+    ASSERT_TRUE(!(*allowed_topics_init == *allowed_topics_final));
+
+    // Delete test file
+    std::filesystem::remove(configfile);
+}
+
+// Test the configuration reload when the yaml configuration file is modified
+TEST(ReloadConfig, yaml)
+{
+    auto configfile = "./file_watcher_test.yaml";
+    write_yaml_file(configfile, 0);
+
+    // Create DDS Enabler
+    std::unique_ptr<DDSEnabler> enabler;
+    bool result = create_dds_enabler(configfile, data_callback, type_callback, nullptr, enabler);
+    ASSERT_TRUE(result);
+
+    // Create DDSEnablerAccessor to access protected configuration
+    auto enabler_accessor = static_cast<DDSEnablerAccessor*>(enabler.get());
+    
+    // Take initial configuration (allowed topics)
+    std::shared_ptr<ddspipe::core::AllowedTopicList> allowed_topics_init;
+    enabler_accessor->get_allowed_topics(allowed_topics_init);
+    
+    // Modify configuration file
+    write_yaml_file(configfile, 1);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    // Take final configuration (allowed topics)
+    std::shared_ptr<ddspipe::core::AllowedTopicList> allowed_topics_final;
+    enabler_accessor->get_allowed_topics(allowed_topics_final);
+
+    ASSERT_TRUE(!(*allowed_topics_init == *allowed_topics_final));
+
+    // Delete test file
+    std::filesystem::remove(configfile);
+}
+
+} /* namespace ddsenabler */
+} /* namespace eprosima */
+
+
+int main(
+        int argc,
+        char** argv)
+{
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
```

Comment:
```suggestion
}

```

Reply:
Review all missing EOF please. I believe this is part of our style, I think uncrustify would fail without them.